src/test/java/itest/GraphQLTest.java

```diff
@@ -69,37 +74,45 @@ void testEnvironmentNodeListing() throws Exception {
         query.put(
                 "query",
                 "query { environmentNodes(filter: { name: \"Custom Targets\" }) { name nodeType"
-                        + " descendantTargets { name nodeType } } }");
+                        + " children { name nodeType } } }");
+
         HttpResponse<Buffer> resp =
                 webClient
                         .extensions()
                         .post("/api/v2.2/graphql", query.toBuffer(), REQUEST_TIMEOUT_SECONDS);
         MatcherAssert.assertThat(
                 resp.statusCode(),
                 Matchers.both(Matchers.greaterThanOrEqualTo(200)).and(Matchers.lessThan(300)));
-        EnvironmentNodesResponse actual =
-                mapper.readValue(resp.bodyAsString(), EnvironmentNodesResponse.class);
 
-        EnvironmentNodes expected = new EnvironmentNodes();
+        TypeReference<EnvironmentNodesResponse> typeRef =
+                new TypeReference<EnvironmentNodesResponse>() {};
+        EnvironmentNodesResponse actual = mapper.readValue(resp.bodyAsString(), typeRef);
+
+        List<DiscoveryNode> expectedChildren = new ArrayList<>();
+        DiscoveryNode expectedChild = new DiscoveryNode();
+        expectedChild.name = "service:jmx:rmi:///jndi/rmi://localhost:0/jmxrmi";
+        expectedChild.nodeType = "JVM";
+        expectedChildren.add(expectedChild);
 
-        EnvironmentNode jdp = new EnvironmentNode();
-        jdp.name = "JDP";
-        jdp.nodeType = "Realm";
+        DiscoveryNode expectedNode = new DiscoveryNode();
+        expectedNode.name = "Custom Targets";
+        expectedNode.nodeType = "Realm";
+        expectedNode.children = expectedChildren;
 
-        jdp.descendantTargets = new ArrayList<>();
-        Node cryostat = new Node();
-        cryostat.name = "service:jmx:rmi:///jndi/rmi://localhost:0/jmxrmi";
-        cryostat.nodeType = "JVM";
-        jdp.descendantTargets.add(cryostat);
+        assertThat(actual.getData().getEnvironmentNodes().size(), is(1));
 
-        Node target = new Node();
-        target.name = "service:jmx:rmi:///jndi/rmi://localhost:0/jmxrmi";
-        target.nodeType = "JVM";
-        jdp.descendantTargets.add(target);
+        DiscoveryNode actualNode = actual.getData().getEnvironmentNodes().get(0);
 
-        expected.environmentNodes = List.of(jdp);
+        assertThat(actualNode.name, is(expectedNode.name));
+        assertThat(actualNode.nodeType, is(expectedNode.nodeType));
+        assertThat(actualNode.children.size(), is(expectedNode.children.size()));
 
-        MatcherAssert.assertThat(actual.data, Matchers.equalTo(expected));
+        for (int i = 0; i < actualNode.children.size(); i++) {
```

Comment:
If the `actualNode.children` and `expectedNode.children` don't have the same `.size()`, in particular if the `actualNode.children` is larger, then I think this loop will terminate with an `IndexOutOfBoundsException`, which would be harder to understand what the problem really was. I think it would be better to figure a way to continue using `Matchers.equalTo()`, since this will provide a nice summary of the differences in the lists even in cases where they are of different lengths. But maybe this needs a custom comparator: https://hamcrest.org/JavaHamcrest/javadoc/2.0.0.0/org/hamcrest/comparator/ComparatorMatcherBuilder.html